wdl-lsp/src/server.rs

```diff
@@ -439,9 +439,31 @@ impl LanguageServer for Server {
         debug!("received `textDocument/didOpen` request: {params:#?}");
 
         if let Ok(path) = params.text_document.uri.to_file_path() {
-            if let Err(e) = self.analyzer.add_directory(path).await {
+            if path.is_file() {
+                // Handle individual files
+                if let Err(e) = self
+                    .analyzer
+                    .add_document(path_to_uri(&path).expect("should convert to uri"))
+                    .await
+                {
+                    error!(
+                        "failed to add document {uri}: {e}",
+                        uri = params.text_document.uri
+                    );
+                    return;
+                }
+            } else if path.is_dir() {
+                // Handle individual directory
```

Comment:
Checked the [SPEC](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocumentItem) once again, do we really need to check for directories as only document uris will be received  ?

Reply:
I'm fairly certain only files will be received and we can error in the case of a directory.

I think we can simply change the original call to `add_document` without having to read the file's metadata and let it error in analysis when opening it if it's not a file.